src/test/java/com/microsoft/sqlserver/jdbc/parametermetadata/ParameterMetaDataCacheTest.java

```diff
@@ -0,0 +1,248 @@
+                    + " [nvarchar](32) COLLATE Latin1_General_BIN2 ENCRYPTED WITH (" + "COLUMN_ENCRYPTION_KEY="
+                    + cekName + ",ENCRYPTION_TYPE=Deterministic," + "ALGORITHM='AEAD_AES_256_CBC_HMAC_SHA_256') NULL,"
+                    + column2 + " [nvarchar](32) COLLATE Latin1_General_BIN2 ENCRYPTED WITH ("
+                    + "COLUMN_ENCRYPTION_KEY=" + cekName + ",ENCRYPTION_TYPE=Randomized,"
+                    + "ALGORITHM='AEAD_AES_256_CBC_HMAC_SHA_256') NULL)");
+
+        }
+    }
+
+    private void updateTable(Connection connection, String tableName, String firstColumn, String secondColumn,
+            String firstData, String secondData) throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            stmt.executeUpdate("declare @firstColumn nvarchar(32) = " + firstData
+                    + " declare @secondColumn nvarchar(32) = " + secondData + " insert into " + tableName + "("
+                    + firstColumn + ", " + secondColumn + ") values (@firstColumn, @secondColumn)");
+        } catch (SQLException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private void selectFromTable(Connection connection, String tableName, String firstColumn,
+            String firstData) throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            stmt.executeUpdate(" select * from " + tableName + " where " + firstColumn + " = " + firstData + ";");
+        } catch (SQLException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    private void alterTable(Connection connection, String tableName, String firstColumn,
+            String newCek) throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            stmt.executeUpdate("alter table " + tableName + " modify " + firstColumn + " encrypt with " + newCek);
+        } catch (SQLException e) {
+            fail(e.getMessage());
+        }
+    }
+    
+    private long timedTestSelect(Connection connection, String tableName, String firstColumn,
+            String firstData) throws SQLException {
+        long timer = System.currentTimeMillis();
+        try (Statement stmt = connection.createStatement()) {
+            stmt.executeUpdate(" select * from " + tableName + " where " + firstColumn + " = " + firstData + ";");
+        } catch (SQLException e) {
+            fail(e.getMessage());
+        }
+        return System.currentTimeMillis() - timer;
+    }
+
+    private void createCMK(String cmkName, String keyPath) throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            String sql = " if not exists (SELECT name from sys.column_master_keys where name='" + cmkName + "')"
+                    + " begin" + " CREATE COLUMN MASTER KEY " + cmkName + " WITH (KEY_STORE_PROVIDER_NAME = '"
+                    + Constants.AZURE_KEY_VAULT_NAME + "', KEY_PATH = '" + keyPath + "')" + " end";

```

Comment:
Should keystore provider be windows key store provider? eg. MSSQL_CERTIFICATE_STORE